# HG changeset patch
# User David Grundberg <davidg@cs.umu.se>
# Date 1269448078 -3600
#      Wed Mar 24 17:27:58 2010 +0100
# Node ID 12d25a1d84bfbebe64d8deb3f21a334250ee85b7
# Parent  ba932ad87a04d93e4571d78515ae04d42ea4eca3
Use gnulib headers instead of C++ std headers where needed.

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,23 @@
+2010-03-24  David Grundberg  <davidg@cs.umu.se>
+
+	* cmd-edit.cc: Include gnulib stdlib.h/string.h instead of C++
+	headers.
+	* file-ops.cc: Include gnulib stdlib.h/stdio.h/string.h instead of
+	C++ headers.
+	* lo-math.h: Include gnulib math.h instead of C++ headers.
+	* lo-utils.cc: Include gnulib stdlib.h instead of C++ headers.
+	* oct-env.cc, dir-ops.cc, file-stat.cc, cmd-hist.cc: Include
+	gnulib string.h instead of C++ headers.
+	* oct-time.cc: Include gnulib time.h instead of C++ headers.
+	* oct-time.h (octave_time): Use default namespace ::modf instead
+	of std::modf.
+	* oct-uname.cc: Include gnulib string.h instead of C++ headers.
+
 2010-03-24  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Sparse.cc (lblookup): New helper func.
 	(Sparse<T>::index (const idx_vector&, bool)): Use it here for lookups.
 	(Sparse<T>::index (const idx_vector&, const idx_vector&, bool)):
 	Specialize for scalar row index.
 
 2010-03-23  John W. Eaton  <jwe@octave.org>
diff --git a/liboctave/cmd-edit.cc b/liboctave/cmd-edit.cc
--- a/liboctave/cmd-edit.cc
+++ b/liboctave/cmd-edit.cc
@@ -20,18 +20,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include <cstdlib>
-#include <cstring>
+#include "stdlib.h"
+#include "string.h"
 
 #include <string>
 
 #include <sys/types.h>
 #include <unistd.h>
 
 #include "quit.h"
 
diff --git a/liboctave/cmd-hist.cc b/liboctave/cmd-hist.cc
--- a/liboctave/cmd-hist.cc
+++ b/liboctave/cmd-hist.cc
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include <cstring>
+#include "string.h"
 
 #include <iostream>
 #include <string>
 
 #include "cmd-edit.h"
 #include "cmd-hist.h"
 #include "file-ops.h"
 #include "lo-error.h"
diff --git a/liboctave/dir-ops.cc b/liboctave/dir-ops.cc
--- a/liboctave/dir-ops.cc
+++ b/liboctave/dir-ops.cc
@@ -25,16 +25,17 @@ along with Octave; see the file COPYING.
 #endif
 
 #include <cerrno>
 #include <cstdlib>
 #include <cstring>
 
 #include <list>
 #include <string>
+#include "string.h"
 
 #include "sysdir.h"
 
 #include "dir-ops.h"
 #include "file-ops.h"
 #include "lo-error.h"
 #include "lo-sysdep.h"
 #include "str-vec.h"
diff --git a/liboctave/file-ops.cc b/liboctave/file-ops.cc
--- a/liboctave/file-ops.cc
+++ b/liboctave/file-ops.cc
@@ -21,19 +21,19 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cerrno>
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
+#include "stdio.h"
+#include "stdlib.h"
+#include "string.h"
 
 #include <iostream>
 #include <vector>
 
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
 
diff --git a/liboctave/file-stat.cc b/liboctave/file-stat.cc
--- a/liboctave/file-stat.cc
+++ b/liboctave/file-stat.cc
@@ -21,17 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cerrno>
-#include <cstring>
+#include "string.h"
 
 #include <sys/types.h>
 #include <unistd.h>
 
 #include "file-ops.h"
 #include "file-stat.h"
 #include "statdefs.h"
 
diff --git a/liboctave/lo-math.h b/liboctave/lo-math.h
--- a/liboctave/lo-math.h
+++ b/liboctave/lo-math.h
@@ -19,22 +19,22 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_liboctave_math_h)
 #define octave_liboctave_math_h 1
 
 #if defined (__cplusplus)
-#include <cmath>
+#include "math.h"
 // if #undef log2 is missing in cmath, undef it here
 #if defined (log2)
 #undef log2
 #endif
 #else
-#include <math.h>
+#include "math.h"
 #endif
 
 #if defined (HAVE_SUNMATH_H)
 #include <sunmath.h>
 #endif
 
 #endif
diff --git a/liboctave/lo-utils.cc b/liboctave/lo-utils.cc
--- a/liboctave/lo-utils.cc
+++ b/liboctave/lo-utils.cc
@@ -22,17 +22,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cctype>
-#include <cstdlib>
+#include "stdlib.h"
 #include <cstdio>
 #include <cstring>
 
 #include <limits>
 #include <string>
 
 #include <sys/types.h>
 #include <unistd.h>
diff --git a/liboctave/oct-env.cc b/liboctave/oct-env.cc
--- a/liboctave/oct-env.cc
+++ b/liboctave/oct-env.cc
@@ -38,17 +38,17 @@ Free Software Foundation, Inc.
 */ 
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cctype>
 #include <cstdlib>
-#include <cstring>
+#include "string.h"
 
 #include <string>
 
 #include <sys/types.h>
 #include <unistd.h>
 
 #include "progname.h"
 
diff --git a/liboctave/oct-time.cc b/liboctave/oct-time.cc
--- a/liboctave/oct-time.cc
+++ b/liboctave/oct-time.cc
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <climits>
-#include <ctime>
+#include "time.h"
 
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 
 #include "strftime.h"
 
 #include "lo-error.h"
diff --git a/liboctave/oct-time.h b/liboctave/oct-time.h
--- a/liboctave/oct-time.h
+++ b/liboctave/oct-time.h
@@ -63,17 +63,17 @@ public:
     ot_usec = rem;
     ot_unix_time += extra;
   }
 
   octave_time (double d)
     : ot_unix_time (static_cast<time_t> (d)), ot_usec (0)
   {
     double ip;
-    ot_usec = static_cast<int> (std::modf (d, &ip) * 1e6);
+    ot_usec = static_cast<int> (::modf (d, &ip) * 1e6);
   }
 
   octave_time (const octave_base_tm& tm);
 
   octave_time (const octave_time& ot)
     : ot_unix_time (ot.ot_unix_time), ot_usec (ot.ot_usec) { }
 
   octave_time& operator = (const octave_time& ot)
diff --git a/liboctave/oct-uname.cc b/liboctave/oct-uname.cc
--- a/liboctave/oct-uname.cc
+++ b/liboctave/oct-uname.cc
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cerrno>
-#include <cstring>
+#include "string.h"
 
 #ifdef HAVE_SYS_UTSNAME_H
 #include <sys/utsname.h>
 #endif
 
 #include "oct-uname.h"
 
 void
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,18 @@
+2010-03-24  David Grundberg  <davidg@cs.umu.se>
+
+	* oct-parse.yy: Include gnulib stdlib.h/stdio.h instead of C++
+	headers.
+	* oct-fstrm.cc: Include gnulib stdlib.h/string.h instead of C++
+	headers.
+	* mex.cc: Include gnulib stdlib.h instead of C++ headers.
+	* input.cc: Include gnulib stdio.h instead of C++ headers.
+	* c-file-ptr-stream.cc: Include gnulib stdio.h.
+
 2010-03-24  John W. Eaton  <jwe@octave.org>
 
 	* load-path.cc (load_path::add_to_fcn_map): Style fixes.
 
 2010-03-24  Jaroslav Hajek  <highegg@gmail.com>
 
 	* variables.cc (maybe_missing_function_hook): Don't call hook if
 	errors are being handled.
diff --git a/src/c-file-ptr-stream.cc b/src/c-file-ptr-stream.cc
--- a/src/c-file-ptr-stream.cc
+++ b/src/c-file-ptr-stream.cc
@@ -21,16 +21,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <iostream>
+#include "stdio.h"
 
 #include "c-file-ptr-stream.h"
 
 #ifndef SEEK_SET
 #define SEEK_SET 0
 #endif
 
 #ifndef SEEK_CUR
diff --git a/src/input.cc b/src/input.cc
--- a/src/input.cc
+++ b/src/input.cc
@@ -22,17 +22,17 @@ along with Octave; see the file COPYING.
 */
 
 // Get command input interactively or from files.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include <cstdio>
+#include "stdio.h"
 #include <cstdlib>
 #include <cstring>
 #include <cassert>
 
 #include <iostream>
 #include <sstream>
 #include <string>
 
diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -20,17 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #include <config.h>
 
 #include <cfloat>
 #include <csetjmp>
 #include <cstdarg>
-#include <cstdlib>
+#include "stdlib.h"
 #include <cstring>
 #include <cctype>
 
 #include <set>
 
 #include "f77-fcn.h"
 #include "lo-ieee.h"
 #include "oct-locbuf.h"
diff --git a/src/oct-fstrm.cc b/src/oct-fstrm.cc
--- a/src/oct-fstrm.cc
+++ b/src/oct-fstrm.cc
@@ -21,17 +21,18 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cerrno>
-#include <cstring>
+#include "string.h"
+#include "stdlib.h"
 
 #include "error.h"
 #include "oct-fstrm.h"
 
 octave_stream
 octave_fstream::create (const std::string& nm_arg, std::ios::openmode arg_md,
                         oct_mach_info::float_format ff)
 {
diff --git a/src/oct-parse.yy b/src/oct-parse.yy
--- a/src/oct-parse.yy
+++ b/src/oct-parse.yy
@@ -29,18 +29,18 @@ along with Octave; see the file COPYING.
 %{
 #define YYDEBUG 1
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cassert>
-#include <cstdio>
-#include <cstdlib>
+#include "stdio.h"
+#include "stdlib.h"
 
 #include <iostream>
 #include <map>
 #include <sstream>
 
 #include "Cell.h"
 #include "Matrix.h"
 #include "cmd-edit.h"
